python/test/test_datasetslice.py

```diff
@@ -0,0 +1,30 @@
+import unittest
+
+from dataset import *
+import numpy as np
+
+class TestDatasetSlice(unittest.TestCase):
+
+    def setUp(self):
+        d = Dataset()
+        d[Coord.X] = ([Dim.X], np.arange(10))
+        d[Data.Value, "a"] = ([Dim.X], np.arange(10)) 
+        d[Data.Value, "b"] = ([Dim.X], np.arange(10)) 
+        self._d = d
+
+    def test_extract_slice(self):
+        ds_slice = self._d["a"]
+        self.assertEqual(type(ds_slice), dataset.DatasetView)
+        # We should have just one data variable
+        self.assertEqual(1, len([var for var in ds_slice if var.is_data]), 1)
```

Comment:
Never seen this, what is this assert with three arguments doing?

Reply:
That's a bug/typo, it's getting interpreted as the message argument apparently. Needs fixing.